tests/test_db_utils.py

```
@@ -726,13 +735,444 @@ def test_schema_upgrade_operational_errors(self):
+            with sqlite3.connect(legacy_db_path1) as conn:
+                conn.execute("CREATE TABLE test_table1 (id INTEGER)")
+
+            os.makedirs(legacy_dir2, exist_ok=True)
+            legacy_db_path2 = os.path.join(legacy_dir2, "meshtastic.sqlite")
+            with sqlite3.connect(legacy_db_path2) as conn:
+                conn.execute("CREATE TABLE test_table2 (id INTEGER)")
+
+            # Set up mocks that will return different legacy dirs on each call
+            call_count = [0]
+
+            def get_legacy_dirs_side_effect(*args, **kwargs):
```

Ruff (0.14.14)

[warning] Unused function argument: `args`

(ARG001)